apps/api_web/lib/api_web/views/trip_view.ex

```diff
@@ -83,6 +98,17 @@ defmodule ApiWeb.TripView do
     )
   end
 
+  def stops(%{id: trip_id, service_id: service_id}, conn) do
+    %{start_date: start_date} = Service.by_id(service_id)
+
+    stop_ids =
+      %{date: start_date, trips: [trip_id]}
+      |> Schedule.filter_by()
+      |> Enum.map(& &1.stop_id)
```

Comment:
- I believe this will always look up the stops, even if they aren't included? You can look at how we do includes for predictions, schedules, or alerts to see how to make it optional
- Rather than looking up the service ID, you should be able to look up the stops directly: `trip_id |> Schedule.by_trip_id() |> Enum.sort_by(& &1.stop_sequence) |> Enum.map(& &1.stop_id)`

Reply:
1. Fixed.
2. I thought we need date to query the schedules, but I've tested your approach and it seems to be working. Fixed.